tokio/src/sync/semaphore.rs

```diff
@@ -73,6 +73,21 @@ use std::sync::Arc;
 /// }
 /// ```
 ///
+/// Limit access to a file:
```

Comment:
I think it would be good to give a more detailed description of the example. For example, it could start with "Limit the number of open files to 100 at the time." Then, it could briefly mention why this is could be useful: Most operating systems have a limit on the number of open files you can have, and trying to open a file when you've reached the limit will return an error. Using a semaphore lets you sleep until another file is closed instead of failing.

Reply:
Makes sense! I have made an attempt to explain why this example could be useful, trying to keep it simple and correct.